### PR TITLE
feat(operator): support autoware operation mode service

### PR DESCRIPTION
**Pull request by TakaHoribe**
## PR Type

- New Feature

## Description

- To support operation_mode interface
- keep vehicle/engage interface for backward compatibility

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [x] All open points are addressed and tracked via issues or tickets
- [x] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s).
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: autoware_iv_internal_api_adaptor/src/operator.cpp [hunk 1]
```diff
@@ -185,6 +190,25 @@ void Operator::setVehicleEngage(bool engage)
   pub_vehicle_engage_->publish(msg);
 }
 
+void Operator::setVehicleOperationMode(uint8_t mode)
```
Comment (isamu-takagi) [inline autoware_iv_internal_api_adaptor/src/operator.cpp]: The failure is not reflected in the response. Set the response like the `setExternalSelect` function.

Comment (TakaHoribe) [reply autoware_iv_internal_api_adaptor/src/operator.cpp]: Thank you. Addressed.

File: autoware_iv_internal_api_adaptor/src/operator.cpp [hunk 2]
```diff
@@ -35,6 +35,8 @@ Operator::Operator(const rclcpp::NodeOptions & options) : Node("external_api_ope
 
   cli_external_select_ = proxy.create_client<tier4_control_msgs::srv::ExternalCommandSelect>(
     "/control/external_cmd_selector/select_external_command");
+  cli_operation_mode_ = proxy.create_client<tier4_system_msgs::srv::OperationModeRequest>(
+    "/system/operation_mode_request");
```
Comment (isamu-takagi) [inline autoware_iv_internal_api_adaptor/src/operator.cpp:39]:
```suggestion
    "/system/control_mode_request");
```

Comment (isamu-takagi) [reply autoware_iv_internal_api_adaptor/src/operator.cpp:39]: https://github.com/autowarefoundation/autoware.universe/pull/1246

**Review by isamu-takagi — APPROVED** [review]
LGTM

**Pull request merged**